test/versioned/fastify/package.json

```diff
@@ -12,34 +17,65 @@
         "fastify": ">=2.0.0 < 3"
       },
       "files": [
-        "add-hook.tap.js",
-        "code-level-metrics-middleware.tap.js",
-        "code-level-metrics-hooks.tap.js",
-        "errors.tap.js",
-        "fastify-2-naming.tap.js",
-        "new-state-tracking.tap.js"
+        "add-hook.test.js",
+        "code-level-metrics-middleware.test.js",
+        "code-level-metrics-hooks.test.js",
+        "errors.test.js",
+        "fastify-2-naming.test.js",
+        "new-state-tracking.test.js"
+      ]
+    },
+
+    {
+      "engines": {
+        "node": ">=18"
+      },
+      "dependencies": {
+        "fastify": "^3.0.0",
+        "middie": ">=3.0.0 && <7.0.0"
+      },
+      "files": [
+        "fastify-gte3.test.js",
+        "fastify-gte3-naming.test.js"
       ]
     },
+
     {
       "engines": {
         "node": ">=18"
       },
       "dependencies": {
-        "fastify": ">=3.0.0",
```

Comment:
we've lost testing any future major versions now. was this actually needed? also this has increased the test matrix a little bit

Reply:
The block below (line 62 in the updated file) tests `fastify@5`. It's going to be a while before v6 shows up, but we _could_ make that one `>=5.0.0`. 

But, yes, this is needed in order to make sure the right version of the "middie" module is installed that matches the version of Fastify under test. The middie modules utilize Fastify's version matching to assert that it is being loaded into a supported version of Fastify. Without these changes, we end up with version mismatches.

Reply:
got it. then I would put `>=5.0.0`. so we don't lose test coverage when a new major version is eventually released

Reply:
Done. But, to be clear, when v6 is released, all of the plugins will also update their compatible versions. So a change here will very likely still be needed.

Reply:
That's fine